src/compiler/checker.ts

```diff
@@ -21809,6 +21809,10 @@ namespace ts {
             return type.flags & TypeFlags.Union ? every((<UnionType>type).types, f) : f(type);
         }
 
+        function everyContainedType(type: Type, f: (t: Type) => boolean): boolean {

```

Comment:
How do we not have a full set of these yet? Uhh......if you switch the preceding function to allow UnionOrIntersection instead of just Union, does everything break? If so, maybe it's ok to silently make the existing function work on intersections.

Reply:
```
    compiler tests
  [▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․]
Test failure:
"before all" hook for "Correct errors for tests/cases/compiler/deepKeysIndexing.ts"
  [▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬․․․․․․․․․․․․․․․․․․․․․․․․․․․]
Test failure:
"before all" hook for "Correct errors for tests/cases/compiler/intersectionsOfLargeUnions.ts"

Test failure:
"before all" hook for "Correct errors for tests/cases/compiler/intersectionsOfLargeUnions2.ts"
  [▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬․․․․․․․․․․․․․․․․․․․․]
Test failure:
"before all" hook for "Correct errors for tests/cases/compiler/narrowingByTypeofInSwitch.ts"
  [▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬․․․․․․․․]
```

I'm going to go ahead and say... probably not! 😄